src/cart/crt_hg.c

```diff
@@ -1720,9 +1722,15 @@ crt_hg_bulk_transfer_cb(const struct hg_cb_info *hg_cbinfo)
 	crt_bulk_cbinfo.bci_bulk_desc = bulk_desc;
 
 	rc = bulk_cbinfo->bci_cb(&crt_bulk_cbinfo);
-	if (rc != 0)
-		D_ERROR("bulk_cbinfo->bci_cb failed, rc: %d.\n", rc);
+	if (rc != 0) {
+		int rc2;
+		Dl_info dlinfo;
 
+		rc2 = dladdr(bulk_cbinfo->bci_cb, &dlinfo);
+		D_ERROR("bulk_cbinfo->bci_cb (%s()) failed, rc: %d.\n",
+			rc2 && dlinfo.dli_saddr == bulk_cbinfo->bci_cb ? dlinfo.dli_sname :
+									 "<unresolved>", rc);
```

Comment:
what is this for?

Reply:
to print real symbol of call-back

Reply:
ok, but how does this relate to the ticket? there are number of callbacks like this, so why are we only doing this one only? its not a bad change on its own, but sounds like it should be done in a separate PR with more callbacks being covered.

Reply:
ok, I understand and agree will push a new commit without this particular change!